Server/src/server/dataCenter/models/card/Deck.java

```diff
@@ -93,47 +86,15 @@ public void removeCard(ServerCard card) throws ClientException {
         if (!hasCard(card.getCardId())) {
             throw new ClientException("deck doesn't have this card.");
         }
-        if (hero == card)
+        if (card.equals(hero)) {
             hero = null;
-        if (item == card)
-            item = null;
-        others.remove(card);
+        }
+        cards.remove(card);
     }
 
-
     public boolean isValid() {
         if (hero == null) return false;
-        return others.size() == 20;
-    }
-
-    public void copyCards() {//TODO:reCode
-        if (hero != null) {
-            this.hero = new ServerCard(hero);
-            this.hero.setCardId(makeId(hero, 1));
-        }
-
-        List<ServerCard> oldOthers = this.others;
-        this.others = new ArrayList<>();
-        for (ServerCard other : oldOthers) {
-            ServerCard card = new ServerCard(other);
-            card.setCardId(makeId(card, numberOf(card.getName()) + 1));
-            others.add(card);
-        }
-    }
-
-    private String makeId(ServerCard card, int number) {
```

Comment:
if this method isnt used, where else are we creating card IDs?

Reply:
Server Collection.addCard and a shared Card constructor. I'm not sure where the best place is but short term I think we're better without unused code.